specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/disk.json

```diff
@@ -919,7 +951,33 @@
           ],
           "x-ms-enum": {
             "name": "DiskState",
-            "modelAsString": true
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Unattached",
+                "description": "The disk is not being used and can be attached to a VM."
+              },
+              {
+                "value": "Attached",
+                "description": "The disk is currently mounted to a running VM."
+              },
+              {
+                "value": "Reserved",
+                "description": "The disk is mounted to a stopped-deallocated VM"
+              },
+              {
+                "value": "ActiveSAS",
+                "description": "The disk currently has an Active SAS Uri associated with it."
+              },
+              {
+                "value": "ReadyToUpload",
+                "description": "The upload disk does not have an active write SAS, but the customer can request on to upload the VHD."
+              },
+              {
+                "value": "ActiveUpload",
+                "description": "The upload disk has an active write SAS associated with it."
```

Comment:
A disk is created for upload and a write token has been issued for uploading to it.